test/integration-test.sh

```diff
@@ -20,7 +20,10 @@ echo "Checking that the ecs-scripts installed correctly"
 configure-ecs-instance --help
 
 echo "Using gruntwork-install to install a module from the gruntwork-install repo and passing args to it via --module-param"
-gruntwork-install --module-name "dummy-module" --repo "https://github.com/gruntwork-io/gruntwork-installer" --tag "v0.0.17" --module-param "file-to-cat=$SCRIPT_DIR/integration-test.sh"
+gruntwork-install --module-name "dummy-module" --repo "https://github.com/gruntwork-io/gruntwork-installer" --tag "v0.0.25" --module-param "file-to-cat=$SCRIPT_DIR/integration-test.sh"
+
+echo "Using gruntwork-install to install a test module from the gruntwork-install repo and test that it's args are maintained via --module-param"
+gruntwork-install --module-name "args-test" --repo "https://github.com/gruntwork-io/gruntwork-installer" --tag "v0.0.25" --module-param "test-args=1 2 3 *"
```

Comment:
OK, so... something a bit odd. I ran this exact command, but pointing to a local copy of `args-test`... And it passes just fine, with none of the changes in this PR. To make local testing easier, the only things I changed were a few temporary hacks to `gruntwork-install`:

1. Comment out the `if ! repo_is_public "$repo"` check.
1. Update `fetch_script_module` to use `cp` instead of `git clone` to "install" the `args-test` module I had locally.

It sounds like you're having issues with argument expansion and whitespace processing, but I'm not able to repro this. Note that I'm on OS X; is there something that behaves differently on Linux or a newer version of Bash?

Reply:
I'll have to play with it some more. It was failing during my packer image builds (Amazon Linux 2). I'll see if I can recreate it with a simpler Docker image

Reply:
Here is a confirmed case where shell escaping is still breaking.

See my branch at jeckhart/gruntwork-installer#je-test-param-escaping.

Checkout the branch and run:
```
export GITHUB_OAUTH_TOKEN=AAAbbb111222
docker build --build-arg GITHUB_OAUTH_TOKEN="${GITHUB_OAUTH_TOKEN}" -t gruntwork/gruntwork-installer-test-param-escaping test/test-param-escaping
```

The docker build should fail with the error:
```
...
...
 ---> Running in d4906c5dde57
+ gruntwork-install --module-name metrics/cloudwatch-memory-disk-metrics-scripts --tag v0.21.0 --repo https://github.com/gruntwork-io/module-aws-monitoring --module-param 'cron-schedule=*/5 * * * *'
...
...
Adding the following crontab to cwmonitoring: */5 bin boot dev etc home install.sh lib lib64 local media mnt opt proc root run sbin srv sys tmp usr var bin boot dev etc home install.sh lib lib64 local media mnt opt proc root run sbin srv sys tmp usr var bin boot dev etc home install.sh lib lib64 local media mnt opt proc root run sbin srv sys tmp usr var * /opt/aws-scripts-mon/mon-put-instance-data.pl --disk-path='/' --mem-util --disk-space-util --auto-scaling --from-cron
"/tmp/cloudwatch-monitoring-crontab":1: bad hour
errors in crontab file, can't install.
The command '/bin/sh -c /install.sh' returned a non-zero code: 1
```

Reply:
And to validate that this PR does fix that test-case, you can run a version that installs gruntwork-installer from the PR and the Dockerfile passes:

```
export GITHUB_OAUTH_TOKEN=AAAbbb111222
docker build --build-arg GITHUB_OAUTH_TOKEN="${GITHUB_OAUTH_TOKEN}" -t gruntwork/gruntwork-installer-test-param-escaping -f test/test-param-escaping/Dockerfile.working test/test-param-escaping
```

Results in:
```
...
...
2020-05-04 17:28:34 [INFO] [gruntwork-install] Executing /tmp/gruntwork-script-modules/metrics/cloudwatch-memory-disk-metrics-scripts/install.sh --cron-schedule */5 * * * *
...
...
Adding the following crontab to cwmonitoring: */5 * * * * /opt/aws-scripts-mon/mon-put-instance-data.pl --disk-path='/' --mem-util --disk-space-util --auto-scaling --from-cron
Cleaning up monitoring cache files in /var/tmp/aws-mon/
2020-05-04 17:28:46 [INFO] [gruntwork-install] Success!
Removing intermediate container baf51d6e9be1
 ---> 2b3318e296ad
Step 10/10 : CMD ["echo", "This container is used for testing. Consider running one of the test scripts under the /test folder."]
 ---> Running in 4c4892dbc0a1
Removing intermediate container 4c4892dbc0a1
 ---> 5eae5f7f8761
Successfully built 5eae5f7f8761
Successfully tagged gruntwork/gruntwork-installer-test-param-escaping:latest
```